data_structures/src/data_request.rs

```diff
@@ -47,10 +46,18 @@ impl DataRequestPool {
     }
 
     /// Get all reveals related to a `DataRequestOuput`
-    pub fn get_reveals(&self, dr_pointer: &Hash) -> Option<Vec<&RevealTransaction>> {
-        self.data_request_pool
-            .get(dr_pointer)
-            .map(|dr_state| dr_state.info.reveals.values().collect())
+    pub fn get_reveals(&self, dr_pointer: &Hash) -> Option<Vec<RevealTransaction>> {
+        self.data_request_pool.get(dr_pointer).map(|dr_state| {
+            let mut pkhs: Vec<PublicKeyHash> = dr_state.info.reveals.keys().cloned().collect();
+
+            pkhs.sort();
+            let reveals: Vec<RevealTransaction> = pkhs
+                .iter()
+                .map(|pkh| dr_state.info.reveals.get(pkh).unwrap().clone())
+                .collect();
```

Comment:
You can avoid calling collect twice because the reveal transactions already include the pkh, so you can sort by key:

```rust
    pub fn get_reveals(&self, dr_pointer: &Hash) -> Option<Vec<&RevealTransaction>> {
        self.data_request_pool.get(dr_pointer).map(|dr_state| {
            let mut reveals: Vec<&RevealTransaction> =
                dr_state.info.reveals.values().collect();
            reveals.sort_unstable_by_key(|reveal| reveal.body.pkh);

            reveals
        })
    }
```

Reply:
But why are we sorting by pkh in the first place? Maybe we can leave this function like it was before, and sort the reveals where we actually need them to be sorted.

Reply:
in both places where use these functions we need that are sorted so... make sense for me that return the result sorted. This sorting is needed to apply tally scripts and punish positions